core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -670,6 +875,14 @@ public void close() {
                 } catch (InterruptedException e) {
                     // ignore
                 }
+                remoteStorageReaderThreadPool.shutdownNow();
+                //waits for 2 mins to terminate the current tasks
+                try {
+                    remoteStorageReaderThreadPool.awaitTermination(2, TimeUnit.MINUTES);
```

Comment:
How did we decide on 2min. here? I don't think we should block shutdown of the broker on this over here because there are other limits associated with clean vs unclean shutdown. If we do plan to block, we should tie it to overall shutdown timeout. As an example, clean shutdown is expected to be completed in 5 min. see `lifecycleManager.controlledShutdownFuture.get(5L, TimeUnit.MINUTES)` in BrokerServer.scala.